src/sentry/incidents/models/incident.py

```diff
@@ -203,6 +203,9 @@ class Incident(Model):
     activation = FlexibleForeignKey(
         "sentry.AlertRuleActivations", on_delete=models.SET_NULL, null=True
     )
+    subscription = FlexibleForeignKey(
+        "sentry.QuerySubscription", null=True, on_delete=models.PROTECT
+    )
```

Comment:
Are we sure we want protect here? I feel like we can recreate the subscription when we update the rule, which will now be blocked if we protect here. Do we want to set null insted?